pallets/ds-maps/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use frame_support::{
+    codec::{Decode, Encode},
+    storage::StorageDoubleMap,
+    sp_runtime::sp_std::ops::{Add, Sub, Div, Mul, Shl, BitOr},
+    decl_error, decl_event, decl_module, decl_storage, dispatch, ensure,    
+    weights::Weight,
+    Parameter,
+};
+use substrate_fixed::types::U9F23;
+
+use frame_system::ensure_signed;
+use pallet_ds_accounts as accounts;
+use accounts::REGISTRAR_ROLE;
+
+mod default_weight;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub const GREEN_AREA: u8 = 0b00000001;
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Copy, Default, Debug, PartialEq, Eq)]
+pub struct Point2D<Coord> {
+    lon: Coord,
+    lat: Coord,
+}
+
+impl<Coord> Point2D<Coord> {
+    pub fn new(lon: Coord, lat: Coord) -> Self {
+        Point2D{lon, lat}
+    }
+}
+
+
+//derives and if req by compiler
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Debug, Clone, PartialEq, Eq)]
+pub struct Rect2D<Point2D> {
+    north_west: Point2D,
+    south_east: Point2D,
+}
+
+impl<Point2D> Rect2D<Point2D> {
+    pub fn new(north_west: Point2D, south_east: Point2D) -> Self {
+        Rect2D{north_west, south_east}
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default)]
+pub struct Zone<ZoneId, Rect2D> {
+    pub zone_id: ZoneId,
+    pub rect: Rect2D,
+    pub height: u16,
+}
+
+impl<ZoneId, Rect2D> Zone<ZoneId, Rect2D> {
+    pub fn new(zone_id: ZoneId, rect: Rect2D, height: u16) -> Self {
+        Zone {zone_id, rect, height}
+    }
+} 
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default, Debug, PartialEq, Eq)]
+pub struct Point3D<Coord> {
+    lat: Coord,
+    lon: Coord,
+    alt: Coord,
+}
+
+impl<Coord> Point3D<Coord> {
+    pub fn new(lat: Coord, lon: Coord, alt: Coord) -> Self {
+        Point3D{lat, lon, alt}
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Debug, Clone, PartialEq, Eq)]
+pub struct Box3D<Point3D> {
```

Comment:
I'd say, the type argument should be `Coord`, and the components should have type `Point3D<Coord>`.
For now, nothing makes this box really 3D. What if I pass, say, `u8` as `Point3D` argument. We'll get not really a box at all...

Reply:
Also applies to all other structs here.